teps/0091-trusted-resources.md

```diff
@@ -364,26 +372,137 @@ type KeylessRef struct {
+			Status:  corev1.ConditionFalse,
+			Reason:  podconvert.ReasonResourceVerificationFailed
+			Message: "", //filled with error message,
+		}
+ ```
+
+### How feature flag and Verification Policy update the status
+* If the `trusted-resources-verification-no-match-policy` is set to `fail`
+  * No Matching Policies found: Resources that do not have matched policies will fail the taskrun/pipelinerun.
+
+  Resulting yaml status:
+  ```yaml
+  status:
+    conditions:
+    - lastTransitionTime: "2023-03-01T18:17:05Z"
+      message: no policies are matched
+      reason: ResourceVerificationFailed
+      status: "False"
+      type: TrustedResourcesVerified
+    - lastTransitionTime: "2023-03-01T18:17:10Z"
+      message: no policies are matched
+      reason: ResourceVerificationFailed
+      status: "False"
+      type: Succeeded
+  ```
+
+   * Matching Policies found:
+     * Don't fail taskrun/pipelinerun if all the matched `enforce` policies pass or only match `warn` policies.
+      Resulting yaml status if all policies pass:
+      ```yaml
+      status:
+        conditions:
+        - lastTransitionTime: "2023-03-01T18:17:05Z"
+          message: Trusted resource verification passed
+          reason: ResourceVerificationSucceeded
+          status: "True"
+          type: TrustedResourcesVerified
+        - lastTransitionTime: "2023-03-01T18:17:10Z"
+          message: All Steps have completed executing
+          reason: Succeeded
+          status: "True"
+          type: Succeeded
+      ```
+      Resulting yaml status with if only `warn` mode policy fails:
+      ```yaml
+      status:
+        conditions:
+        - lastTransitionTime: "2023-03-01T18:17:05Z"
+          message: Trusted resource verification failed
+          reason: ResourceVerificationFailed
+          status: "False"
+          type: TrustedResourcesVerified
+        - lastTransitionTime: "2023-03-01T18:17:10Z"
+          message: All Steps have completed executing
+          reason: Succeeded
+          status: "True"
+          type: Succeeded
+      ```
+     * Fail the taskrun/pipelinerun if any matched `enforce` policy fails to verify.
+      Resulting yaml status with `enforce` mode policy, note that the run will fail:
+      ```yaml
+      status:
+        conditions:
+        - lastTransitionTime: "2023-03-01T18:17:05Z"
+          message: Trusted resource verification failed
+          reason: ResourceVerificationFailed
+          status: "False"
+          type: TrustedResourcesVerified
+        - lastTransitionTime: "2023-03-01T18:17:10Z"
+          message: resource verification failed
+          reason: ResourceVerificationFailed
+          status: "False"
+          type: Succeeded
+      ```
+
+ * If the `trusted-resources-verification-no-match-policy` is set to `ignore` or `warn`,
+   * No Matching Policies found: Don't fail taskrun/pipelinerun, skip the verification if set to `ignore`, and emit warning log if set to `warn`.
```

Comment:
so there's no `TrustedResourcesVerified` in this case? if so, please make it explicit